tests/concurrency.py

```diff
@@ -19,13 +22,23 @@
     await asyncio.sleep(seconds)
 
 
-try:
-    import trio
-    from httpx.concurrency.trio import TrioBackend
-except ImportError:  # pragma: no cover
-    pass
-else:
+@sleep.register(TrioBackend)
+async def _sleep_trio(backend, seconds: int):
+    await trio.sleep(seconds)
+
+
+@functools.singledispatch
+async def run_concurrently(backend, *async_fns):
+    raise NotImplementedError  # pragma: no cover
+
+
+@run_concurrently.register(AsyncioBackend)
+async def _run_concurrently_asyncio(backend, *async_fns):
+    await asyncio.gather(*(fn() for fn in async_fns))
```

Comment:
Nit: I find this slightly hard to read, could we unpack this?

Reply:
Sure!